classes/migration/upgrade/v3_4_0/PreflightCheckStatsMigration.inc.php

```diff
@@ -0,0 +1,179 @@
+                DB::table('metrics')->where('submission_id', '=', $submissionId)->delete();
+            }
+
+            // Clean orphaned metrics submission file IDs
+            $orphanedIds = DB::table('metrics AS m')->leftJoin('submission_files AS sf', 'm.assoc_id', '=', 'sf.submission_file_id')->where('m.assoc_type', '=', Application::ASSOC_TYPE_SUBMISSION_FILE)->whereNull('sf.submission_file_id')->distinct()->pluck('m.assoc_id');
+            foreach ($orphanedIds as $submissionFileId) {
+                $this->_installer->log("Removing orphaned metrics submission file ID {$submissionFileId}.");
+                DB::table('metrics')->where('assoc_type', '=', Application::ASSOC_TYPE_SUBMISSION_FILE)->where('assoc_id', '=', $submissionFileId)->delete();
+            }
+            // Clean orphaned metrics submission supp file IDs
+            $orphanedIds = DB::table('metrics AS m')->leftJoin('submission_files AS sf', 'm.assoc_id', '=', 'sf.submission_file_id')->where('m.assoc_type', '=', Application::ASSOC_TYPE_SUBMISSION_FILE_COUNTER_OTHER)->whereNull('sf.submission_file_id')->distinct()->pluck('m.assoc_id');
+            foreach ($orphanedIds as $submissionSuppFileId) {
+                $this->_installer->log("Removing orphaned metrics submission supplementary file ID {$submissionSuppFileId}.");
+                DB::table('metrics')->where('assoc_type', '=', Application::ASSOC_TYPE_SUBMISSION_FILE_COUNTER_OTHER)->where('assoc_id', '=', $submissionSuppFileId)->delete();
+            }
+
+            // Clean orphaned metrics representation IDs
+            $orphanedIds = DB::table('metrics AS m')->leftJoin('publication_galleys AS r', 'm.representation_id', '=', 'r.galley_id')->whereNotNull('m.representation_id')->whereNull('r.galley_id')->distinct()->pluck('m.representation_id');
+            foreach ($orphanedIds as $representationId) {
+                $this->_installer->log("Removing orphaned metrics article galley ID {$representationId}.");
+                DB::table('metrics')->where('representation_id', '=', $representationId)->delete();
+            }
+
+            // m.pkp_section_id will not be considered here, because it will be not migrated
+
+            // Inform about probably old and not anymore supported assoc_type in the DB table metrics
+            $oldAssocTypes = DB::table('metrics AS m')->whereNotIn('m.assoc_type', [Application::getContextAssocType(), Application::ASSOC_TYPE_ISSUE, Application::ASSOC_TYPE_ISSUE_GALLEY, Application::ASSOC_TYPE_SUBMISSION, Application::ASSOC_TYPE_SUBMISSION_FILE, Application::ASSOC_TYPE_SUBMISSION_FILE_COUNTER_OTHER])->distinct()->pluck('m.assoc_type');
+            foreach ($oldAssocTypes as $oldAssocType) {
+                $this->_installer->log("The DB table metrics contains old and not anymore supported assoc_type {$oldAssocType} that will be not migrated to new metrics tables.");
```

Comment:
Because they will not be dropped -- they will stay in the metrics table that will be renamed to metrics_old -- also the other metric types (other than ojs::counter) will stay there -- so I just removed that part of the sentence...